docs/guides/postgres/backup/kubestash/auto-backup/index.md

```diff
@@ -0,0 +1,863 @@
+metadata:
+  name: gcs-storage
+  namespace: demo
+spec:
+  storage:
+    provider: gcs
+    gcs:
+      bucket: kubestash-qa
+      prefix: blueprint
+      secretName: gcs-secret
+  usagePolicy:
+    allowedNamespaces:
+      from: All
+  default: true
+  deletionPolicy: WipeOut
+```
+
+Let's create the BackupStorage we have shown above,
+
+```bash
+$ kubectl apply -f https://github.com/kubedb/docs/raw/{{< param "info.version" >}}/docs/guides/postgres/backup/kubestash/auto-backup/examples/backupstorage.yaml
+backupstorage.storage.kubestash.com/gcs-storage created
+```
+
+Now, we are ready to backup our database to our desired backend.
+
+**Create RetentionPolicy:**
+
+Now, let's create a `RetentionPolicy` to specify how the old Snapshots should be cleaned up.
+
+Below is the YAML of the `RetentionPolicy` object that we are going to create,
+
+```yaml
+apiVersion: storage.kubestash.com/v1alpha1
+kind: RetentionPolicy
+metadata:
+  name: demo-retention
+  namespace: demo
+spec:
+  default: true
+  failedSnapshots:
+    last: 2
+  maxRetentionPeriod: 2mo
+  successfulSnapshots:
+    last: 5
+  usagePolicy:
+    allowedNamespaces:
+      from: All
+```
+
+Let’s create the above `RetentionPolicy`,
+
+```bash
+$ kubectl apply -f https://github.com/kubedb/docs/raw/{{< param "info.version" >}}/docs/guides/postgres/backup/kubestash/auto-backup/examples/retentionpolicy.yaml
+retentionpolicy.storage.kubestash.com/demo-retention created
+```
+
+**Create Secret:**
+
+We also need to create a secret with a `Restic` password for backup data encryption.
+
+Let's create a secret called `encrypt-secret` with the Restic password,
+
+```bash
+$ echo -n 'changeit' > RESTIC_PASSWORD
+$ kubectl create secret generic -n demo encrypt-secret \
+    --from-file=./RESTIC_PASSWORD \
+secret "encrypt-secret" created
+```
+
+## Auto-backup with default configurations
+
+In this section, we are going to backup a `PostgreSQL` database of `demo` namespace. We are going to use the default configurations which will be specified in the `BackupBlueprint` CR.
+
+**Prepare Backup Blueprint**
+
+A `BackupBlueprint` allows you to specify a template for the `Repository`,`Session` or `Variables` of `BackupConfiguration` in a Kubernetes native way.
+
+Now, we have to create a `BackupBlueprint` CR with a blueprint for `BackupConfiguration` object.
+
+```yaml
+apiVersion: core.kubestash.com/v1alpha1
+kind: BackupBlueprint
+metadata:
+  name: postgres-default-backup-blueprint
+  namespace: demo
+spec:
+  usagePolicy:
+    allowedNamespaces:
+      from: All
+  backupConfigurationTemplate:
+    deletionPolicy: OnDelete
+    # ============== Blueprint for Backends of BackupConfiguration  =================
+    backends:
+      - name: gcs-backend
+        storageRef:
+          namespace: demo
+          name: gcs-storage
+        retentionPolicy:
+          name: demo-retention
+          namespace: demo
+    # ============== Blueprint for Sessions of BackupConfiguration  =================
```

Comment:
We have only `BackupConfiguration` Blueprint in case of kubestash. Remove the comments.